tests/ga/test_policy_engine.py

```diff
@@ -15,20 +15,447 @@
+
+    def test_should_not_enforce_signature_if_global_signatureRequired_false_and_no_individual_policy(self):
+        """
+        If signatureRequired is false globally and no individual extension signature policy, should_enforce_signature_validation = False.
+        """
+        test_extension = Extension(name=TEST_EXTENSION_NAME)
+        policy = \
+            {
+                "policyVersion": "0.1.0",
+                "extensionPolicies": {
+                    "allowListedExtensionsOnly": True,
+                    "signatureRequired": False,
+                    "extensions": {}
+                }
+            }
+        self._create_policy_file(policy)
+        engine = ExtensionPolicyEngine()
+        should_enforce_signature = engine.should_enforce_signature_validation(test_extension)
+        self.assertFalse(should_enforce_signature,
+                            msg="Global signatureRequired policy is false, so signature should not be enforced.")
+
+    def test_should_enforce_signature_if_global_signatureRequired_true_and_individual_signatureRequired_not_specified(self):
+        """
+        If individual policy is present, but signatureRequired is not specified for that policy, use global signatureRequired.
+        """
+        test_extension = Extension(name=TEST_EXTENSION_NAME)
+        policy = \
+            {
+                "policyVersion": "0.1.0",
+                "extensionPolicies": {
+                    "allowListedExtensionsOnly": True,
+                    "signatureRequired": True,
+                    "extensions": {
+                        TEST_EXTENSION_NAME: {}
+                    }
+                }
+            }
+        self._create_policy_file(policy)
+        engine = ExtensionPolicyEngine()
+        should_enforce_signature = engine.should_enforce_signature_validation(test_extension)
+        self.assertTrue(should_enforce_signature,
+                            msg="Individual signatureRequired policy is not set, so should use global policy and enforce signature.")
+
+    def test_should_not_enforce_signature_if_no_custom_policy_present(self):
+        test_extension = Extension(name=TEST_EXTENSION_NAME)
+        engine = ExtensionPolicyEngine()
+        should_enforce_signature = engine.should_enforce_signature_validation(test_extension)
+        self.assertFalse(should_enforce_signature,
+                         msg="No custom policy is present, so use default policy. Should not enforce signature.")
+
+    def test_should_allow_if_no_custom_policy_present(self):
+        test_extension = Extension(name=TEST_EXTENSION_NAME)
+        engine = ExtensionPolicyEngine()
+        should_allow = engine.should_allow_extension(test_extension)
+        self.assertTrue(should_allow,
+                        msg="No custom policy is present, so use default policy. Should allow all extensions.")
+
+    def test_should_allow_if_extension_policy_section_missing(self):
```

Comment:
Can we add a case to test that we don't enforce signature if extension policy section is missing

Reply:
Added